crates/runtime/src/dataconnector/mod.rs

```diff
@@ -297,6 +297,14 @@ pub enum DataConnectorError {
         dataset_name: String,
         differences: String,
     },
+
+    #[snafu(display(
+        "The dataset name '{keyword}' is a protected keyword for the {dataconnector} data connector and cannot be used as a name for a dataset.\nPlease change the name of your dataset in your Spicepod."
```

Comment:
```suggestion
        "The name '{keyword}' cannot be used as a name for a dataset for the {dataconnector} data connector.\nChange the name in the Spicepod and try again."
```

Please refer to https://github.com/spiceai/spiceai/blob/trunk/docs/dev/error_handling.md for our error handling guidelines.